.github/workflows/semgrep.yml

```diff
@@ -15,4 +15,7 @@ jobs:
     if: (github.actor != 'dependabot[bot]')
     steps:
       - uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
-      - uses: returntocorp/semgrep-action@245bf11ddb2f3d4e35f116608cf6e27ae0f9aa04 # v1
+      - uses: returntocorp/semgrep-action@8689a93eca50c91f8e8cfa61e3c4fb7e96d12bf2 # v1
+permissions:
+  contents: read
+
```

Comment:
We're making a lot of changes in this PR to existing actions that have nothing to do with the CircleCI migration. There's no reasoning why in the PR description (which I expect we'll update with human-authored description at some point?). I don't think any of this belongs here.

Reply:
Hi Tim, these changes were added as part of the "Restrict Workflow Permissions" work item on the GHA migration [ticket](https://github.com/hashicorp/GHA-migration/issues/241). It was an automated change introduced by the bot and has been done in all migrated repositories to date.

We can certainly link back to that ticket if you'd like too, is there some kind of details or information you are looking to see in the description?

Reply:
That the internal ticket is poorly scoped to include unrelated work isn't really relevant.  Each chunk of work should be in its own PR, just as we would for any other work on Nomad.

This is an open source project and changes should be as transparent as possible to contributors -- not to mention Nomad Engineering contributors, for whom that link or statement of work had no link to this PR.

Reply:
I updated the PR description to clearly state the goals, both primary (migration) and secondary (standardization), and the resulting changes.

I agree with you in general about scope, but I'm hoping the description (which will be commit-message'd too after squash) is now sufficiently clear for us and for our open source community, which I take to be your main concern, laudably.  How's it look to ya?

Reply:
Alrighty, we removed all the non-workflow code here (and I updated the description to point to those PRs for posterity).  How's it look to ye now?